File: yashandb/src/test/java/com/dbx/agent/yashandb/YashandbAgentTest.java
package com.dbx.agent.yashandb;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.JdbcAgentProfile;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import java.util.Collections;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class YashandbAgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new YashandbAgent();
    }

    @Override
    protected String resultSetSql() {
        return "SELECT 1 FROM DUAL";
    }

    @Test
    void declaresYashandbJdbcProfile() {
        JdbcAgentProfile profile = YashandbAgent.YASHANDB_PROFILE;

        Assertions.assertEquals("com.yashandb.jdbc.Driver", profile.getDriverClass());
        Assertions.assertEquals("jdbc:yasdb://{host}:{port}/{database}", profile.getUrlTemplate());
        Assertions.assertTrue(profile.getSkipExecutionContext());
        Assertions.assertEquals(Collections.emptySet(), profile.getExcludedSchemas());
    }
}

File: yashandb/src/test/kotlin/com/dbx/agent/yashandb/YashandbAgentTest.kt
package com.dbx.agent.yashandb

import com.dbx.agent.DatabaseAgent
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class YashandbAgentTest : JdbcFakeExecutionBehaviorTest() {
    override fun createAgent(): DatabaseAgent {
        return YashandbAgent()
    }

    override fun resultSetSql(): String = "SELECT 1 FROM DUAL"

    @Test
    fun `declares yashandb jdbc profile`() {
        val profile = YashandbAgent().profile

        assertEquals("com.yashandb.jdbc.Driver", profile.driverClass)
        assertEquals("jdbc:yasdb://{host}:{port}/{database}", profile.urlTemplate)
        assertTrue(profile.skipExecutionContext)
        assertEquals(setOf("SYS", "MDSYS", "XA_SYS"), profile.excludedSchemas)
    }
}
